workflows/test_validate.py

```diff
@@ -0,0 +1,130 @@
+import mock
+import workflows
+from workflows.validate import validate_recipe, main
+import json
+import sys
+
+
+def test_no_file():
+    """Test that validate returns an error when it returns with no file"""
+    with pytest.raises(Exception):
+        validate_recipe()
+
+
+def test_healthy_recipe():
+    """Test that no errors are raised when given a healthy recipe"""
+    healthy_recipe = """
+        {
+          "1": {
+            "queue": "simpleservice.submission",
+            "parameters": {
+              "commands": [
+                "echo This is a command32"
+              ],
+              "workingdir": "/dls/tmp/riw56156/zocalo",
+              "output_file": "out.txt"
+            }
+          },
+          "start": [
+            [1, []]
+          ]
+        }
+    """
+
+    # Run validate with mock open, expect no exceptions
+    with mock.patch(
+        "builtins.open", mock.mock_open(read_data=healthy_recipe), create=True
+    ) as mock_file:
+        validate_recipe("/path/to/open")
+
+
+def test_bad_json():
+    """Test that no errors are raised when given a bad json recipe"""
+    bad_json = """
+        {
+          "1": {
+            "queue": "simpleservice.submission"
+            "parameters": {
+              "commands": [
+                "echo This is a command32"
+              ],
+              "workingdir": "/dls/tmp/riw56156/zocalo",
+              "output_file": "out.txt"
+            }
+          },
+          "start": [
+            [1, []]
+          ]
+        }
+    """
+
+    # Run validate with mock open, expect JSON error
+    with pytest.raises(json.decoder.JSONDecodeError):
+        with mock.patch(
+            "builtins.open", mock.mock_open(read_data=bad_json), create=True
+        ) as mock_file:
+            validate_recipe("/path/to/open")
+
+
+def test_bad_recipe():
+    """Test that no errors are raised when given a bad zocalo recipe"""
+    bad_recipe = """
+        {
+          "1": {
+            "queue": "simpleservice.submission",
+            "parameters": {
+              "commands": [
+                "echo This is a command32"
+              ],
+              "workingdir": "/dls/tmp/riw56156/zocalo",
+              "output_file": "out.txt"
+            }
+          }
+        }
+    """
+
+    # Run validate with mock open, expect JSON error
+    with pytest.raises(workflows.Error):
+        with mock.patch(
+            "builtins.open", mock.mock_open(read_data=bad_recipe), create=True
+        ) as mock_file:
+            validate_recipe("/path/to/open")
+
+
+# Create a mock of the validate call
+@mock.patch("workflows.validate.validate_recipe")
+def test_main(mock_requests):
+    """Test that the main function call works with one inputs"""
+    # Create fake arguments to test on
+    test_args = ["validate.py", "file1"]
+    with mock.patch.object(sys, "argv", test_args):
+        main()
+    mock_requests.assert_called_once()
+    mock_requests.assert_called_with("file1")
```

Comment:
there is `.assert_called_once_with(...)`

Reply:
Done